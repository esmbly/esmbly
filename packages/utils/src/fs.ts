import fs from 'fs';
import util from 'util';

export default {
  readdir: util.promisify(fs.readdir),
  readFile: util.promisify(fs.readFile),
  stat: util.promisify(fs.stat),
  writeFile: util.promisify(fs.writeFile),
};
