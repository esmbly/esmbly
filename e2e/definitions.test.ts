import fs from 'fs';
import util from 'util';
import path from 'path';
import { testRunner } from './helpers/testRunner';

const readFile = util.promisify(fs.readFile);
const example = path.resolve(path.join('examples', 'definitions'));
const types = path.join(example, 'types');

describe('E2E: definitions', () => {
  beforeAll(async () => {
    jest.setTimeout(10000);
    await testRunner(example);
  });

  it('outputs TypeScript definition files', async () => {
    const index = await readFile(path.join(types, 'index.d.ts'), 'utf8');
    const add = await readFile(path.join(types, 'utils', 'add.d.ts'), 'utf8');
    const multiply = await readFile(
      path.join(types, 'utils', 'multiply.d.ts'),
      'utf8',
    );
    expect(index).toMatchSnapshot();
    expect(add).toMatchSnapshot();
    expect(multiply).toMatchSnapshot();
  });
});
