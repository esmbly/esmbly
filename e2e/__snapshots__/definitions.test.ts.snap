// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2E: definitions outputs TypeScript definition files 1`] = `
"/**
 * Perform operations on a number
 * @param {number} n
 * @returns {object}
 */
export declare const operateOn: (n: number) => object;
"
`;

exports[`E2E: definitions outputs TypeScript definition files 2`] = `
"/**
 * Add two numbers
 * @param {number} a
 * @param {number} b
 * @returns {number}
 */
export declare function add(a: number, b: number): number;
"
`;

exports[`E2E: definitions outputs TypeScript definition files 3`] = `
"/**
 * Multiply two numbers
 * @param {number} a
 * @param {number} b
 * @returns {number}
 */
export declare function add(a: number, b: number): number;
"
`;
